recurring_contract/views/recurring_contract_view.xml

```diff
@@ -31,38 +32,36 @@
             <form>
                 <header>
                     <field name="state" widget="statusbar"/>
-                    <button name="contract_waiting" class="oe_highlight" string="Validate" states="draft" type="object"/>
+                    <button name="contract_waiting" class="oe_highlight" string="Validate" states="draft"
+                            type="object"/>
                     <button name="action_cancel_draft" type="object" string="Set back to draft" states="cancelled"/>
                     <button name="%(action_end_contract)d" type="action" string="Terminate" class="oe_highlight"
                             attrs="{'invisible': ['|', ('state', 'in', ['terminated', 'cancelled']), ('end_date', '!=', False)]}"/>
                 </header>
                 <sheet>
                     <div class="oe_button_box" name="button_box">
-                        <button name="button_generate_invoices" type="object" class="oe_inline oe_stat_button"
-                                icon="fa-refresh" string="Generate invoices" states="waiting,active"/>
```

Comment:
I'm not sure it's good to remove the button from the contract. It means that the user will always need to go in the group when he wants to generate invoices (one click more). This button could simply be a shortcut calling the same method. I think users would complain if the button is removed.

Reply:
I agree with you but putting that button on the contract is quite confusing because from the contract you'll generate invoices for the group. So for me we could put a button on the partner in addition of the contract group but we shouldn't put it on the contract.

Reply:
It has been like that forever and no one was ever confused. The partner view already has hundred of buttons, I think it's already a confused view where I don't wan't to include every possible action. In general, when working with sponsorships, we are in the contract view and it's from there that we need the invoices. Contract group is a rarely accessed view.